spring-data-couchbase/src/main/java/org/springframework/data/couchbase/core/convert/CryptoConverter.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2022 the original author or authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.couchbase.core.convert;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.nio.charset.StandardCharsets;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.core.convert.ConversionService;
+import org.springframework.data.convert.PropertyValueConverter;
+import org.springframework.data.convert.ValueConversionContext;
+import org.springframework.data.couchbase.core.mapping.CouchbaseDocument;
+import org.springframework.data.couchbase.core.mapping.CouchbasePersistentEntity;
+import org.springframework.data.couchbase.core.mapping.CouchbasePersistentProperty;
+import org.springframework.data.mapping.PersistentProperty;
+import org.springframework.util.Assert;
+
+import com.couchbase.client.core.encryption.CryptoManager;
+import com.couchbase.client.java.json.JsonObject;
+
+/**
+ * Encrypt/Decrypted properties annotated with
+ * 
+ * @author Michael Reiche
+ */
+public class CryptoConverter implements
+		PropertyValueConverter<Object, CouchbaseDocument, ValueConversionContext<? extends PersistentProperty<?>>> {
+
+	CryptoManager cryptoManager;
+
+	public CryptoConverter(CryptoManager cryptoManager) {
+		this.cryptoManager = cryptoManager;
+	}
+
+	@Override
+	public Object read(CouchbaseDocument value, ValueConversionContext<? extends PersistentProperty<?>> context) {
+		byte[] decrypted = cryptoManager().decrypt(value.export());
+		if (decrypted == null) {
+			return null;
+		}
+		// it's decrypted into a String. Now figure out how to convert to the property type.
+		CouchbaseConversionContext ctx = (CouchbaseConversionContext) context;
+		CouchbasePersistentProperty property = ctx.getProperty();
+		org.springframework.data.convert.CustomConversions conversions = ctx.getConverter().getConversions();
+		ConversionService svc = ctx.getConverter().conversionService;
+
+		boolean wasString = false;
+		List<Exception> exceptionList = new LinkedList<>();
+		String decryptedString = new String(decrypted);
```

Comment:
I'm not really happy with the following code that takes the decrypted string and turns it into a simple type.  But it seems to do the job.